### ISSUE TITLE
[bug] Error: Specify the 'user' and the 'channel' or neither of them when using version ranges

### ISSUE THREAD

**Issue by ojnick** [conan-io/conan#11764]
Starting yesterday we've started running into an issue with using version ranges when referencing dependencies. This has been working for over a year and has suddenly started causing issues. When we attempt to use the command:

`conan install .. -u` (where `..` contains a conanfile.py) we get the error `ERROR: Specify the 'user' and the 'channel' or neither of them. [Remote: myremote]`

I've seen mixed results without the `-u` and may depend on if we need to go out to the remote or not.

Some testing has seemed to isolate the issue to be somehow related to the configured remote. If I disable my remotes (conancenter is still enabled), it works as expected. We are using jfrog Artifactory for our internal conan repos. I tested with a brand new, empty conan repo on jfrog as well as our production servers with the same issue.

May be related to #11762 but I'm not sure. Found #10913 but starting from a fresh cache did not fix my issue.

I had conan version 1.44.1 on my machine and saw the issue; upgraded to 1.51.0 and still see the issue. Experiencing it on Windows and Ubuntu 18.04.

Willing to help dig into this with some guidance as this is a critical issue for us. The workaround is to not use version ranges which we can use in some cases but not many.

### Steps to reproduce (Include if Applicable)

1. Create 2 packages such that one depends on the other (App --> Lib) 
2. Create multiple releases for the Lib package with a user and channel
    * lib/1.0.0@user/channel
    * lib/1.0.1@user/channel
    * lib/1.1.0@user/channel
    * lib/2.0.0@user/channel
    * lib/2.1.0@user/channel
3. Specify a version range App (`lib/[~1]@user/channel`)
4. Add a remote other than conancenter
5. Make a subfolder in App
   * App/
       * conanfile.py
       * build/
6. `cd` into `build` folder and run `conan install -u ..` 
    * See ` Error: Specify the 'user' and the 'channel' or neither of them`?

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10 and Ubuntu 18.04
  * Compiler+version: msvc, gcc
  * Conan version: 1.44.1, 1.51.0   
  * Python version: 3.9.6, 3.6.9

**Comment by czoido** [comment]
Hi @ojnick,

I'm not able to reproduce the issue, that error (`Specify the 'user' and the 'channel' or neither of them`) usually happens when you are adding a requires that has the user but not the channel or viceversa, like for example `requires = "lib/[~1]@user"` (note the missing channel)
Would it be possible to copy the actual `conanfile.py `that you are using for the `conan install` ?

By the way, do you know which Artifactory version you are using?

Thanks a lot.

**Comment by ojnick** [comment]
I've attached my 2 test `conanfile.py` files I used for testing/reproduction. They're extremely minimal. 

Here's a list of the cache:
![image](https://user-images.githubusercontent.com/108154631/182635427-2e9a3c4d-1c61-4248-929a-5df9ad90841f.png)

Here's the list of remotes:
![image](https://user-images.githubusercontent.com/108154631/182635693-2951a26e-7596-4dc6-a622-bf6cd709dfc1.png)

If I disable the remotes that are not `conancenter` it seems to work fine. Having any of the other remotes in place I get the error. The are both repos on Jfrog Cloud. I'm not sure how to get a version from there.

[conanfile _app.py.txt](https://github.com/conan-io/conan/files/9251957/conanfile._app.py.txt)
[conanfile_lib.py.txt](https://github.com/conan-io/conan/files/9251961/conanfile_lib.py.txt)

**Comment by ojnick** [comment]
I came in this morning and after sending my previous comment I tested it again and the issue has seemed to have gone away. Is it possible something with communication with the server (on the server side) got broken and then fixed?

**Comment by czoido** [comment]
Hi @ojnick,

I'm happy to hear that it's working now but no idea what could have happened there, nothing on our side changed. Could it be somehow related to [this](https://github.com/conan-io/conan/issues/10913#issuecomment-1083356588)?. 

Please, report if it happens again.

Thanks a lot.

**Comment by ojnick** [comment]
@czoido I/we've never touched the `storage.path` setting. The issue was also across multiple users and machines with no changes. Another commenter on that issue mentioned wiping the conan `data` folder. I tried that by moving `conan home` and still had the issue, but only when I added in my jfrog cloud remote.

Everything seems to be working again somehow, so I'll close this issue for now. If we experience it again, I'll re-open.
